SimPEG/survey.py

```diff
@@ -1,204 +1,522 @@
+#         Parameters
+#         ----------
+#         instance : class
+#             The class used to validate the input argument *value*
+#         value :
+#             The input used to define the locations for a given receiver.
 
+#         Returns
+#         -------
+#         properties.Array
+#             The receiver location array
+#         """
+#         value = np.atleast_2d(value)
+#         return super(RxLocationArray, self).validate(instance, value)
 
-class SourceLocationArray(properties.Array):
 
-    class_info = "a 1D array denoting the source location"
+# class SourceLocationArray(properties.Array):
+#     """Locations array for sources"""
 
-    def validate(self, instance, value):
-        if not isinstance(value, np.ndarray):
-            value = np.atleast_1d(np.array(value))
-        return super(SourceLocationArray, self).validate(instance, value)
+#     class_info = "a 1D array denoting the source location"
 
+#     def validate(self, instance, value):
+#         """Validation method for setting locations array
 
-class BaseRx(properties.HasProperties):
-    """SimPEG Receiver Object"""
+#         Parameters
+#         ----------
+#         instance : class
+#             The class used to validate the input argument *value*
+#         value :
+#             The input used to define the locations for a given source.
 
-    # TODO: write a validator that checks against mesh dimension in the
-    # BaseSimulation
-    # TODO: location
-    locations = RxLocationArray(
-        "Locations of the receivers (nRx x nDim)", shape=("*", "*"), required=True
-    )
+#         Returns
+#         -------
+#         properties.Array
+#             The source location array
+#         """
+#         if not isinstance(value, np.ndarray):
+#             value = np.atleast_1d(np.array(value))
+#         return super(SourceLocationArray, self).validate(instance, value)
 
-    # TODO: project_grid?
-    projGLoc = properties.StringChoice(
-        "Projection grid location, default is CC",
-        choices=["CC", "Fx", "Fy", "Fz", "Ex", "Ey", "Ez", "N"],
-        default="CC",
-    )
 
-    # TODO: store_projections
-    storeProjections = properties.Bool(
-        "Store calls to getP (organized by mesh)", default=True
-    )
+class BaseRx:
+    """Base SimPEG receiver class.
 
-    _uid = properties.Uuid("unique ID for the receiver")
+    Parameters
+    ----------
+    locations : (n_loc, ndim) numpy.ndarray
+        Locations assocated with a given receiver
+    storeProjections : bool, Default = ``False``
+        Store projections from the mesh to receiver
+    """
+
+    _Ps = None
 
-    _Ps = properties.Dictionary(
-        "dictonary for storing projections",
-    )
 
-    def __init__(self, locations=None, **kwargs):
-        super(BaseRx, self).__init__(**kwargs)
-        if locations is not None:
+    def __init__(self, locations=None, storeProjections=False, **kwargs):
+
+        if locations is None:
```

Comment:
best to not have locations as a kwarg here and change it to an argument, since it's actually required. Python will allow the person interacting with the code to still use it as a kwarg.